diff --git a/lib/services/codepipeline.d.ts b/lib/services/codepipeline.d.ts
index 3339c6f3cbec691690155c07f31123814d0cea3f..5f5b5cff5507fa08c6c77a337069fc2f6a945359 100644
--- a/lib/services/codepipeline.d.ts
+++ b/lib/services/codepipeline.d.ts
@@ -757,8 +757,8 @@ export interface TransitionState {
 export declare const TransitionState: S.Schema<TransitionState>;
 export interface ActionRevision {
     revisionId: string;
-    revisionChangeId: string;
-    created: Date;
+    revisionChangeId?: string;
+    created?: Date;
 }
 export declare const ActionRevision: S.Schema<ActionRevision>;
 export type ActionExecutionStatus = "InProgress" | "Abandoned" | "Succeeded" | "Failed" | (string & {});
diff --git a/lib/services/codepipeline.js b/lib/services/codepipeline.js
index be407d6e5f03315f2eae355b71a13fa66a45fe59..82fd2a75cc26033bbe89db24aec2bbe7b4245dda 100644
--- a/lib/services/codepipeline.js
+++ b/lib/services/codepipeline.js
@@ -624,8 +624,8 @@ export const TransitionState = /*@__PURE__*/ S.suspend(() => S.Struct({
 });
 export const ActionRevision = /*@__PURE__*/ S.suspend(() => S.Struct({
     revisionId: S.String,
-    revisionChangeId: S.String,
-    created: S.Date.pipe(T.TimestampFormat("epoch-seconds")),
+    revisionChangeId: S.optional(S.String),
+    created: S.optional(S.Date.pipe(T.TimestampFormat("epoch-seconds"))),
 })).annotate({ identifier: "ActionRevision" });
 export const ActionExecutionStatus = /*@__PURE__*/ S.String;
 export const ErrorDetails = /*@__PURE__*/ S.suspend(() => S.Struct({ code: S.optional(S.String), message: S.optional(S.String) })).annotate({ identifier: "ErrorDetails" });
diff --git a/src/services/codepipeline.ts b/src/services/codepipeline.ts
index c959a8675db82b1dc6b58c79b23a431920407339..ee0dbacaefb8e4be0e779159424271bfc4872c2f 100644
--- a/src/services/codepipeline.ts
+++ b/src/services/codepipeline.ts
@@ -1628,14 +1628,14 @@ export const TransitionState = /*@__PURE__*/ S.suspend(() =>
 }) as any as S.Schema<TransitionState>;
 export interface ActionRevision {
   revisionId: string;
-  revisionChangeId: string;
-  created: Date;
+  revisionChangeId?: string;
+  created?: Date;
 }
 export const ActionRevision = /*@__PURE__*/ S.suspend(() =>
   S.Struct({
     revisionId: S.String,
-    revisionChangeId: S.String,
-    created: S.Date.pipe(T.TimestampFormat("epoch-seconds")),
+    revisionChangeId: S.optional(S.String),
+    created: S.optional(S.Date.pipe(T.TimestampFormat("epoch-seconds"))),
   }),
 ).annotate({ identifier: "ActionRevision" }) as any as S.Schema<ActionRevision>;
 export type ActionExecutionStatus =
